Components/Sensors/ThermocoupleTask.cpp

```diff
@@ -0,0 +1,331 @@
+
+	*///------------------------------------------------------------------------------
+
+	//Storable Data ------------------------------------------------------------------------------
+
+
+	SOAR_PRINT("\n-- Sample Thermocouple Data --\n");
+
+	uint8_t tempDataBuffer5[5] = {0};
+	//See Above bit mem-map
+
+	//Read ---------------------------------------------------------------------------------------
+	HAL_GPIO_WritePin(TC1_NCS_GPIO_Port, TC1_NCS_Pin, GPIO_PIN_SET);
+	HAL_GPIO_WritePin(TC2_NCS_GPIO_Port, TC2_NCS_Pin, GPIO_PIN_SET);
+
+
+    //Read From Thermocouple 1 first
+	HAL_GPIO_WritePin(TC1_NCS_GPIO_Port, TC1_NCS_Pin, GPIO_PIN_RESET); //begin read with CS pin low
+	HAL_Delay(10);
+	HAL_SPI_Receive(SystemHandles::SPI_Thermocouple1, tempDataBuffer5, 5, 1000); //Fill the data buffer with data from TC1
+	HAL_Delay(10);
+	HAL_GPIO_WritePin(TC1_NCS_GPIO_Port, TC1_NCS_Pin, GPIO_PIN_SET); //end read with setting CS pin to high again
+
+	for(int i = 0; i<5; i++){
+		if(i!=4)
+		dataBuffer1[i] = tempDataBuffer5[i];\
+	}
+
+	int Temp1=0;
+
+	if(!(dataBuffer1[1] & 0x01)){ //if there is not an error with TC1 compute temperature
+		if((dataBuffer1[0]&(0x80))>>7==1)  // Negative Temperature
+		{
+			Temp1 = (dataBuffer1[0] << 6) | (dataBuffer1[1] >> 2);
+			Temp1&=0b01111111111111;
+			Temp1^=0b01111111111111;
+			temperature1 = (((double)-Temp1 / 4)*100-320);
+		}
+		else  // Positive Temperature
+		{
+			Temp1 = (dataBuffer1[0] << 6) | (dataBuffer1[1] >> 2);
+			temperature1 = (((double)Temp1 / 4)*100-320);
+		}
+	}
+	else
+	{
+		temperature1 = 999; //there is an error detected with TC1
+	}
+
+
+
+	tempDataBuffer5[5] = {0}; //reset for TC2 reading
+	//See Above bit mem-map
+
+	//Read ---------------------------------------------------------------------------------------
+	HAL_GPIO_WritePin(TC1_NCS_GPIO_Port, TC1_NCS_Pin, GPIO_PIN_SET);
+	HAL_GPIO_WritePin(TC2_NCS_GPIO_Port, TC2_NCS_Pin, GPIO_PIN_SET);
+
+	//Read From Thermocouple 1 first
+	HAL_GPIO_WritePin(TC2_NCS_GPIO_Port, TC2_NCS_Pin, GPIO_PIN_RESET); //begin read with CS pin low
+	HAL_Delay(10);
+	HAL_SPI_Receive(SystemHandles::SPI_Thermocouple1, tempDataBuffer5, 5, 1000); //Fill the data buffer with data from TC1
+	HAL_Delay(10);
+	HAL_GPIO_WritePin(TC2_NCS_GPIO_Port, TC2_NCS_Pin, GPIO_PIN_SET); //end read with setting CS pin to high again
+
+	for(int i = 0; i<5; i++){
+		if(i!=4)
+		dataBuffer2[i] = tempDataBuffer5[i];
+	}
+
+	int Temp2=0;
+
+	if(!(dataBuffer2[1] & 0x01)){ //if there is not an error with TC1 compute temperature
+		if((dataBuffer2[0]&(0x80))>>7==1)  // Negative Temperature
+		{
+			Temp2 = (dataBuffer2[0] << 6) | (dataBuffer2[1] >> 2);
+			Temp2&=0b01111111111111;
+			Temp2^=0b01111111111111;
+			temperature2 = (((double)-Temp2 / 4)*100-320);
+		}
+		else  // Positive Temperature
+		{
+			Temp2 = (dataBuffer2[0] << 6) | (dataBuffer2[1] >> 2);
+			temperature2 = (((double)Temp2 / 4)*100-320);
+		}
+	}
+	else
+	{
+		temperature2 = 999; //there is an error detected with TC2
```

Comment:
Make this a define or constexpr just at top of ThermocoupleTask.hpp or cpp your choice.